pyccel/ast/numpyext.py

```diff
@@ -563,8 +563,15 @@ def __init__(self, arg):
         if not isinstance(arg, PyccelAstNode):
             raise TypeError('Unknown type of  %s.' % type(arg))
         super().__init__(arg)
-        self._dtype = arg.dtype
-        self._precision = get_final_precision(arg)
+        if isinstance(arg.dtype, (NativeBool, NativeInteger)):
+            self._dtype = NativeInteger()
+            self._precision = 8
+        elif isinstance(arg.dtype, NativeFloat):
+            self._dtype = NativeFloat()
+            self._precision = default_precision[str(arg.dtype)]
+        elif isinstance(arg.dtype, NativeComplex):
+            self._dtype = NativeComplex()
+            self._precision = default_precision[str(arg.dtype)]
```

Comment:
Are you sure about the precision here? Numpy has a `np.float128` type. When I use `np.sum` on an array of this type I get a `np.float128`.
Maybe it should be:
```suggestion
        if isinstance(arg.dtype, NativeBool):
            self._dtype = NativeInteger()
        else:
            self._dtype = arg.dtype
        self._precision = max(arg.precision, default_precision[str(self._dtype)]
```

Reply:
This is much better, thank you! Ideally, we would like `precision` to be one larger than `arg.precision` to avoid overflow issues. This applies to all precisions except the largest one if each datatype. I will fix this.

Reply:
No. We would like the precision to match what is done in numpy. They can decide what is best adapted. Imagine if the precision increased every time you did a calculation, you would soon run out of memory!

Reply:
@EmilyBourne Is this point resolved?

Reply:
Yes I just left the conversation open so @jalalium could see my reply